### ISSUE TITLE
Bitmap.Save 'A generic error occurred in GDI+

### ISSUE THREAD

**Issue by grahamehorner** [dotnet/corefx#42294]
when calling bitmap.Save with a file path where the folder doesn't exist the save method throws a 
System.Runtime.InteropServices.ExternalException: 'A generic error occurred in GDI+.'

IMHO the the folder should be created (to match folder file operations) or throw an expection of DirectoryNotFoundException

**Comment by danmosemsft** [comment]
Seems reasonable - do you want to offer a PR? We would want the same behavior on Unix and Windows, but if the directory was created in the .NET code, that would hopefully be true.

**Comment by safern** [comment]
Seems reasonable to me as well.

**Comment by korsaire** [comment]
I can look into that.

**Comment by Backhage** [comment]
As a user I would expect an exception being thrown if I supplied a non-existing path. Creating folders is a side effect that might hide errors and lead to unexpected behavior.
Please consider changing the current exception to a DirectoryNotFoundException rather than trying to create the full path.

**Comment by danmosemsft** [comment]
What do other API do?

**Comment by grahamehorner** [comment]
@danmosemsft File.Create creates the folder/directory structure

**Comment by korsaire** [comment]
@Backhage I agree with that. If the API change is approved, maybe later an optional bool param can be passed to indicate whether to create a non-existing path - but IMO it's out of scope of this issue.

**Comment by Backhage** [comment]
@grahamehorner At least for me, using .NET Core 3.0 on Windows, File.Create(@"C:\Temp\NonExistingPath\File.tmp"); throws a DirectoryNotFoundException.

**Comment by grahamehorner** [comment]
@Backhage interesting that differs to my experiance with .NET 2.2 and yes your correct, .NET 3.0 throws the exception as I would have expected
